apps/web/src/server/applications/create/case/applications-create-case.controller.js

```diff
@@ -375,32 +303,31 @@ export async function viewApplicationsCreateCaseTeamEmail(req, response) {
  * @type {import('@pins/express').RenderHandler<ApplicationsCreateCaseTeamEmailProps,
  * {}, ApplicationsCreateCaseTeamEmailBody, {}, DomainParams>}
  */
-export async function updateApplicationsCreateCaseTeamEmail(
-	{ session, body, errors: validationErrors },
-	response
-) {
-	const { applicationId } = response.locals;
+export async function updateApplicationsCreateCaseTeamEmail(request, response) {
+	const { properties, updatedApplicationId } = await formatUpdateCaseTeamEmail(
+		request,
+		response.locals
+	);
 
-	if (body.caseEmail) {
-		const values = { caseEmail: body.caseEmail };
-		const payload = bodyToPayload(body);
-
-		const { errors: apiErrors, id: updatedApplicationId } = await updateApplicationDraft(
-			applicationId,
-			payload
-		);
-
-		if (validationErrors || apiErrors || !updatedApplicationId) {
-			return response.render('applications/create/case/_team-email', {
-				errors: validationErrors || apiErrors,
-				values
-			});
-		}
+	if (properties && (properties?.errors || !updatedApplicationId)) {
+		return handleErrors(properties, teamEmailLayout, response);
 	}
 
-	const nextPath = getSessionCaseHasNeverBeenResumed(session)
+	const nextPath = getSessionCaseHasNeverBeenResumed(request.session)
 		? 'applicant-information-types'
 		: 'applicant-organisation-name';
 
-	response.redirect(`/applications-service/create-new-case/${applicationId}/${nextPath}`);
+	response.redirect(`/applications-service/create-new-case/${updatedApplicationId}/${nextPath}`);
 }
+
+/**
+ * View the sector choice step of the application creation
+ *
+ * @param {Record<string, any>} properties
+ * @param {FormCaseLayout} layout
+ * @param {*} response
```

Comment:
the point of this function is to be as abstract as possible, therefore the interface of the properties is very generic